### ISSUE TITLE
Administrator trying to add/update object _queryState

### ISSUE THREAD

**Issue by chrhol** [Starcounter/Home#258]
Tested in the latest version of v2.3 (built locally today so no exact version)

From what I've seen there are two different exceptions in the log. In the first a property in object `_queryState` is updated (and sent as patch). In the second excecption, the whole `_queryState` object is sent.

Is `_queryState` something client-specific and should be ignored? Who should look at it? 
 cc @Mihaiii, @andwah, @miyconst 

See below for exceptions and how to reproduce.

**1**

```
Starcounter.XSON.JsonPatchException: Unknown property in path. Patch: '{"op":"replace","path":"/Administrator_0/Databases/0/_queryState/sqlQuery","value":"select * from Starcounter.Metadata.RawView where LayoutHandle = 63"}'. Property: '_queryState'.
   at Starcounter.XSON.JsonProperty.EvalutateCurrent(ViewModelVersion version) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonProperty.cs:line 163
   at Starcounter.XSON.JsonProperty.<>c__DisplayClass13_0.<DoEvaluate>b__0() in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonProperty.cs:line 83
   at Starcounter.Advanced.XSON.JsonExtension.Scope(Json json, Action action) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Advanced\JsonExtension.cs:line 69
   at Starcounter.XSON.JsonProperty.DoEvaluate(Json root) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonProperty.cs:line 88
   at Starcounter.XSON.JsonProperty.Evaluate(JsonPointer pointer, Json root) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonProperty.cs:line 62
   at Starcounter.XSON.DefaultPatchHandler.Handle(Json root, JsonPatchOperation patchOp, JsonPointer pointer, String value) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\DefaultPatchHandler.cs:line 21
   at Starcounter.XSON.JsonPatch.HandleOnePatch(Json root, JsonPatchOperation op, JsonPointer ptr, String value, Int32 patchCount, ChangeLog changeLog, Object sourceJson, Boolean strictRejection, JsonPatchStatus previousStatus) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 605
   at Starcounter.XSON.JsonPatch.Apply(Json root, JsonTextReader reader, Boolean strictPatchRejection, Object sourceJson, Int32& patchCount) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 526
   at Starcounter.XSON.JsonPatch.Apply(Json root, String patch, Boolean strictPatchRejection, Int32& patchCount) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 414
   at Starcounter.XSON.JsonPatch.Apply(Json root, String patch, Boolean strictPatchRejection) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 379
   at Administrator.Server.Managers.ServerManager.<>c.<RegisterFullModelApi>b__6_2(String data, WebSocket ws) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.Administrator\Server\Managers\ServerManager.cs:line 143
   at Starcounter.Internal.WsGroupInfo.DetermineAndRunHandler(WebSocket ws) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.Internal\Http\WsManager.cs:line 125
   at Starcounter.Internal.AllWsGroups.RunHandler(UInt16 managedHandlerId, UInt32 groupId, WebSocket ws) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.Internal\Http\WsManager.cs:line 288
   at Starcounter.GatewayHandlers.HandleWebSocket(UInt16 managedHandlerId, Byte* rawChunk, BMX_TASK_INFO* taskInfo, Boolean* isHandled) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter\Apps\GatewayHandlers.cs:line 796
Patch={"op":"replace","path":"/Administrator_0/Databases/0/_queryState/sqlQuery","value":"select * from Starcounter.Metadata.RawView where LayoutHandle = 63"}
CurrentProperty=_queryState
Severity=1
HResult=-2146233088
```

**2**
```
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: {. Path '[0].value', line 1, position 72.
   at Newtonsoft.Json.JsonTextReader.ReadStringValue(ReadType readType)
   at Newtonsoft.Json.JsonTextReader.ReadAsString()
   at Starcounter.XSON.JsonPatch.Apply(Json root, JsonTextReader reader, Boolean strictPatchRejection, Object sourceJson, Int32& patchCount) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 476
   at Starcounter.XSON.JsonPatch.Apply(Json root, String patch, Boolean strictPatchRejection, Int32& patchCount) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 414
   at Starcounter.XSON.JsonPatch.Apply(Json root, String patch, Boolean strictPatchRejection) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.XSON\Sessions\JSON-Patch\JsonPatch.cs:line 379
   at Administrator.Server.Managers.ServerManager.<>c.<RegisterFullModelApi>b__6_2(String data, WebSocket ws) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.Administrator\Server\Managers\ServerManager.cs:line 143
   at Starcounter.Internal.WsGroupInfo.DetermineAndRunHandler(WebSocket ws) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.Internal\Http\WsManager.cs:line 125
   at Starcounter.Internal.AllWsGroups.RunHandler(UInt16 managedHandlerId, UInt32 groupId, WebSocket ws) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter.Internal\Http\WsManager.cs:line 288
   at Starcounter.GatewayHandlers.HandleWebSocket(UInt16 managedHandlerId, Byte* rawChunk, BMX_TASK_INFO* taskInfo, Boolean* isHandled) in c:\users\christian\code\Starcounter\Versions\v2.3\Level1\src\Starcounter\Apps\GatewayHandlers.cs:line 796
LineNumber=1
LinePosition=72
Path=[0].value
HResult=-2146233088
```

For the second stacktrace the patch looks like this: `[{"op":"add","path":"/Administrator_0/Databases/0/_queryState","value":{"sqlQuery":"","columns":[],"rows":[],"limitedResult":false}}]`

**How to reproduce**
It doesn't matter which apps are run, just make sure one database is started.

**1** 

- Go to http://localhost:8181/#/databases/default/sql (exchange `default` with the name of your db.
- Refresh the page.
- Write anything in the query-field.
- Exception no 1 happens in `scadminserver` (can be seen in the Starcounter log).

**2**

- Go to http://localhost:8181/#/databases (it's important that the page is refreshed, so an F5 to make sure might be in order)
- Select a running database and click the  `Sql` option.
- Exception no 2 happens in `scadminserver.exe` (can be seen in the Starcounter log)

**Comment by Mihaiii** [comment]
Related with  https://github.com/Starcounter/Home/issues/239 .
I can filter them [like this](https://github.com/Starcounter/Home/issues/239#issuecomment-329964671) (that code is for demo purposes - final version, if will be applied, will have to update `addedPaths` in case an already added path in `addedPaths` is replaced), but I don't think the fix/solution should be applied on that level, like I mentioned in the comment.

**Comment by Mihaiii** [comment]
Please also note that this problem have different side effect:
for example, if from `http://localhost:8181/#/databases/default/sql` (assuming there is a `default` db), you write anything in the query field and then click on the home icon, you won't be able to start/stop an app at this moment (user has to refresh the page in order to do that).

**Comment by Mihaiii** [comment]
@alshakero Please let me know if I should do what I proposed in https://github.com/Starcounter/Home/issues/239#issuecomment-329964671

**Comment by alshakero** [comment]
`this.serverStatus.filterLocalChange = op => !op.path.startWith('$$');`

Should be enough.

**Comment by Mihaiii** [comment]
I did it with     
`this.serverStatus.filterLocalChange = op => !op.path.includes("$$");`
to include cases like `"/Administrator_0/Databases/0/$$hashKey"` and now there's no error (at least related with this topic :) )

The thing is that I can't reproduce https://github.com/Starcounter/Home/issues/258#issuecomment-333094535 with https://github.com/Starcounter/Home/issues/239#issuecomment-329964671 , but I can reproduce it with `this.serverStatus.filterLocalChange = op => !op.path.includes("$$");`

**Comment by alshakero** [comment]
I see. Could you make it
```js
this.serverStatus.filterLocalChange = op => {
console.log('mihaiii', op);
return !op.path.startWith('/$$');
}```
Reproduce the issue, and show me your log?

**Comment by miyconst** [comment]
@atifwaqar also complained about this error, has to be fixed.

**Comment by Mihaiii** [comment]
I apologize for the late response
SC version 2.3.1.7352

Please see the attached file
[localhost-1506696181965.log](https://github.com/Starcounter/Home/files/1344719/localhost-1506696181965.log)
.

**Comment by Mihaiii** [comment]
Closing as done in https://github.com/Starcounter/level1/commit/c956c7b982268d0b5d2aeede33781c183bd85221 .
The fix should be available starting with RC 2.3.1.7371.
